udfs/remote_udfs/nlp/README.md

```diff
@@ -0,0 +1,100 @@
+# Calling Google's Natural Language API Sentiment Analysis Example 
+
+Google's Natural Language API Sentiment Analysis inspects the given text and identifies the prevailing emotional opinion within the text, especially to determine a writer's attitude as positive, negative, or neutral.  
+
+To run this example, you will need the following APIs enabled:
+* [BigQuery API](https://cloud.google.com/bigquery/docs)
+* [Cloud Functions API](https://cloud.google.com/functions/docs)
+* [BigQuery Connections API](https://cloud.google.com/bigquery/docs/working-with-connections#enable_the_connection_service)
+* [Natural Language API](https://cloud.google.com/natural-language/docs/setup#api)
+
+## Running the example
+
+### Creating your BigQuery connection 
+
+To create your BigQuery connection, you need to specify the following:
+* project_id - the project id that you wish to create the connection in
+* location - the location for the external connection in BigQuery
+* connection_name - the name of the connection 
+* display_name - (optional) the name (friendly name) you want to show 
+
+```
+bq mk --connection --display_name=\'$DISPLAY_NAME\' --connection_type=CLOUD_RESOURCE \
+      --project_id=$PROJECT --location=$LOCATION $CONNECTION_NAME
+```
+
+### Getting the service account associated with the connection
+
+You need to grant the connection service account to be able to invoke functions.  
+To do this, first obtain the service account. 
+* location - the location for the external connection in BigQuery
+
+```
+bq ls --connection --location=$LOCATION
+```
+
+Within properties you'll find *serviceAccountId* which will have the service ID you'll need in a subsequent step.
+
+### Granting the service account invoker permissions on the functions
+
+Grant the service account obtained above permissions to invoke the functions.
+* project - the project where your external connection was created
+* service_account - the service account obtained above
+```
+gcloud projects add-iam-policy-binding $PROJECT \
+    --member=serviceAccount:$SERVICE_ACCOUNT --role=roles/cloudfunctions.invoker
```

Comment:
This will grant permission for the service account on project level. Probably better to grant the permission on the function level

gcloud --project=$PROJECT functions add-iam-policy-binding $CF_NAME --member=serviceAccount:$SERVICE_ACCOUNT --role=roles/cloudfunctions.invoker

To do that, probably need to move the deploying your Cloud Function section before Creating your BigQuery connection?

Reply:
I'll update the call to be function level - but I think it has to come after right? we need the service account from the connection

Reply:
(I moved it to after I created the function if that's what you meant)

Reply:
Updated and pushed